src/MongoDB/ReadPreference.c

```diff
@@ -91,6 +91,10 @@ static bool php_phongo_readpreference_init_from_hash(php_phongo_readpreference_t
 
 	if ((maxStalenessSeconds = zend_hash_str_find(props, "maxStalenessSeconds", sizeof("maxStalenessSeconds") - 1))) {
 		if (Z_TYPE_P(maxStalenessSeconds) == IS_LONG) {
+			if (mongoc_read_prefs_get_mode(intern->read_preference) == MONGOC_READ_PRIMARY) {
+				phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT, "%s initialization requires \"maxStalenessSeconds\" field to not be present with \"primary\" mode", ZSTR_VAL(php_phongo_readpreference_ce->name));
+				goto failure;
+			}
```

Comment:
Do we allow `-1` or `0` to be specified for `maxStalenessSeconds` as an "unset" value?

Reading https://github.com/mongodb/specifications/blob/master/source/server-selection/server-selection.rst#maxstalenessseconds, an error is only thrown for a positive value with mode `primary`. Obviously, any positive value less than `smallestMaxStalenessSeconds` would be rejected, which is why I'm asking about handling a negative/zero for "unset".

It looks like our construct has a guard for `MONGOC_NO_MAX_STALENESS` (as do the functions that populate properties), so I wonder if we should add that somewhere above this block as well for consistency. This is admittedly a nit, as normal operation wouldn't trigger this unless users were crafting their own serialization or set_state stuff. Perhaps this can go in the condition for `Z_TYPE_P(maxStalenessSeconds) == IS_LONG` with logical OR?

Reply:
That `if` won't work, as the `else` path throws an exception. I've added a separate check for `MONGOC_NO_MAX_STALENESS` instead.